code/game/turfs/change_turf.dm

```diff
@@ -114,6 +114,24 @@ GLOBAL_LIST_INIT(blacklisted_automated_baseturfs, typecacheof(list(
 
 	return W
 
+/turf/open/ChangeTurf(path, list/new_baseturfs, flags)
+	if (flags & CHANGETURF_INHERIT_AIR)
+		SSair.remove_from_active(src)
+		var/stashed_air = air
+		air = null // so that it doesn't get deleted
+		. = ..()
+		if (!. || . == src) // changeturf failed or didn't do anything
+			air = stashed_air
+			return
+		if(istype(., /turf/open))
```

Comment:
You should do this check at the beginning since you know what it's changing into